cmd/dep/status.go

```diff
@@ -191,6 +192,25 @@ func (out *dotOutput) MissingHeader()                {}
 func (out *dotOutput) MissingLine(ms *MissingStatus) {}
 func (out *dotOutput) MissingFooter()                {}
 
+type templateOutput struct {
+	w    io.Writer
+	tmpl *template.Template
+}
+
+func (out *templateOutput) BasicHeader() {}
+func (out *templateOutput) BasicFooter() {}
+
+func (out *templateOutput) BasicLine(bs *BasicStatus) {
+	out.tmpl.Execute(out.w, bs)
+}
+
+func (out *templateOutput) MissingHeader() {}
+func (out *templateOutput) MissingFooter() {}
+
+func (out *templateOutput) MissingLine(ms *MissingStatus) {
+	out.tmpl.Execute(out.w, ms)
```

Comment:
Using the same template for `BasicLine` and `MissingLine` makes hard to write a template that handles both.
I think that `-f` should only provide formatting for `BasicLine`.

Reply:
@dolmen thanks for reviewing :)

I don't think there is any use-case for `MissingStatus` with `templateOutput` because `MissingStatus` is used only while printing the error invoked by input-digest mismatch due to missing packages. This function exists because it's required to implement the `outputter` interface and avoid empty output when user tries to use templateout and there's a inputs-digest mismatch due to missing packages.

A non-templateOutput status with missing packages error looks like this.
```
$ dep status
Lock inputs-digest mismatch due to the following packages missing from the lock:

PROJECT                MISSING PACKAGES
github.com/pkg/errors  [github.com/pkg/errors]

This happens when a new import is added. Run `dep ensure` to install the missing packages.
input-digest mismatch
```

When the user does `$ dep status -f="some-template"` only the table output in the above error message would be replaced by the template rendered content. I'm not sure if that can be useful rendered content in any way.

Is there some use-case that I'm overlooking? 🤔